google/auth/_service_account_info.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2016 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Helper functions for loading data from a Google service account file."""
+
+import io
+import json
+
+from google.auth import crypt
+
+
+def from_dict(data, require=None):
+    """Validates a dictionary containing Google service account data and
+    creates a :class:`google.auth.crypt.Signer` instance.
```

Comment:
This is a case where the constraints of a one-liner being on one line don't hurt you.

```python
    """Validates a dictionary containing Google service account data.

    Creates a :class:`google.auth.crypt.Signer` instance corresponding to the data
    in the dictionary.

    ...
    """
```